code/control/DNRoot.php

```diff
@@ -885,6 +887,60 @@ public function getDeployForm() {
 	}
 
 	/**
+	 * @param array $data
+	 * @param DeployForm $form
+	 *
+	 * @return string
+	 */
+	public function showDeploySummary(array $data, DeployForm $form) {
+
+		// Performs canView permission check by limiting visible projects
+		$project = $this->getCurrentProject();
+		if(!$project) {
+			return $this->project404Response();
+		}
+
+		// Performs canView permission check by limiting visible projects
+		$environment = $this->getCurrentEnvironment($project);
+		if(!$environment) {
+			return $this->environment404Response();
+		}
+
+		// get the selected build
+		$selectedBuild =$form->getSelectedBuild($data);
+
+		// clear out the previous deploy from the 'choose revision' view and set required fields
+		$form->setFields(new FieldList(
+             new HiddenField('SelectRelease', 'SelectRelease', 'SHA'),
+             new HiddenField('SHA', 'SHA', $selectedBuild)
+         ));
+
+		// clear out the old action button(s) from the 'choose revision' view
+		$form->setActions(new FieldList());
+
+		// ask the environments backend to amend the deploy form so that
+		// they can later be passed back via the resque job
+		$environment->Backend()->setDeployStrategies($form, $environment);
```

Comment:
I think @micmania1  should review this, because he is working on that bit, unless he is not. I think the $form was removed from the planDeploy?